src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs

```diff
@@ -266,7 +266,7 @@ private bool IsMatchInBypassList(Uri input)
         /// </summary>
         public Uri GetProxy(Uri uri)
         {
-            return uri.Scheme == Uri.UriSchemeHttp ? _httpProxyUri : _httpsProxyUri;
+            return HttpUtilities.IsSupportedNonSecureScheme(uri.Scheme) ? _httpProxyUri : _httpsProxyUri;
```

Comment:
Hi @daddyman

we can use something like below instead of HttpUtilities.IsSupportedNonSecureScheme

uri.Scheme != Uri.UriSchemeWs ? _httpProxyUri : _httpsProxyUri

**Note** : We've to make below items public instead of internal as high lighted below in Uri.cs(codebase linked attached).

![image](https://user-images.githubusercontent.com/5323493/74760418-61f65500-52a0-11ea-84ed-d5865de465ba.png)

Code Base: https://github.com/dotnet/runtime/blob/91c1d7cab87653fc2214d9effbee54d3d9072c65/src/libraries/System.Private.Uri/src/System/Uri.cs

Happy to Pair with you on this :)